health/health.go

```diff
@@ -161,16 +186,46 @@ type warnOptFunc func(*Warnable)
 
 func (f warnOptFunc) mod(w *Warnable) { f(w) }
 
-// Warnable is a health check item that may or may not be in a bad warning state.
-// The caller of NewWarnable is responsible for calling Tracker.SetWarnable to update the state.
+// WarnableCode is a string that distinguishes each Warnable from others. It is globally unique within
+// the program.
+type WarnableCode string
+
+// A Warnable is something that we might want to warn the user about, or not. A Warnable is either
+// in an healthy or unhealth state. A Warnable is unhealthy if the Tracker knows about a WarningState
+// affecting the Warnable.
+// In most cases, Warnables are components of the backend (for instance, "DNS" or "Magicsock").
+// Warnables are similar to the Subsystem type previously used in this package, but they provide
+// a unique identifying code for each Warnable, along with more metadata that makes it easier for
+// a GUI to display the Warnable in a user-friendly way.
 type Warnable struct {
-	debugFlag string // optional MapRequest.DebugFlag to send when unhealthy
+	// Code is a string that uniquely identifies this Warnable across the entire Tailscale backend,
+	// and can be mapped to a user-displayable localized string.
+	Code WarnableCode
+	// Title is a string that the GUI uses as title for any message involving this Warnable.
+	Title string
+	// Text is a function that generates an extended string that the GUI will display when it doesn't know how to
+	// generate a localized string for the Warnable's Code.
+	Text func(args map[string]string) string
```

Comment:
As you suggested I don't think there is much value in doing so, because these values are all ever going to be used as part of string formatting. It would also make the clients much more ugly because you would end up with an JSON object where you cannot know the contained types in advance. In Swift that would result in some very ugly type casting to make it work.